File: packages/python/staffkm-sdk/staffkm/resources/plugins.py
"""Plugins resource."""
from __future__ import annotations

import httpx


class PluginsResource:
    def __init__(self, http: httpx.Client):
        self._http = http

    def list(self) -> list[dict]:
        r = self._http.get("/api/v1/plugins")
        r.raise_for_status()
        return r.json().get("data") or []

    def reload(self) -> dict:
        r = self._http.post("/api/v1/plugins/reload")
        r.raise_for_status()
        return r.json()

File: packages/ts/staffkm-sdk/src/resources/plugins.ts
import { AxiosInstance } from 'axios'

export class PluginsResource {
  constructor(private http: AxiosInstance) {}

  async list(): Promise<any[]> {
    const r = await this.http.get('/api/v1/plugins')
    return r.data?.data ?? []
  }

  async reload(): Promise<any> {
    const r = await this.http.post('/api/v1/plugins/reload')
    return r.data
  }
}
